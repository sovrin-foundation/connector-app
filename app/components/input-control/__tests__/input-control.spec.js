// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import ControlInput from '../input-control'

describe('<WalletTabSendDetails />', () => {
  let onChangeText = jest.fn()
  let validation = jest.fn()
  function setup() {
    const component = renderer.create(
      <ControlInput
        onChangeText={onChangeText}
        placeholder="test"
        label="To"
        name="paymentTo"
        multiline={false}
        validation={validation}
        isValid={'IDLE'}
      />
    )
    const instance = component.root.findByType(ControlInput).instance
    return { component, instance }
  }
  it('should render properly and match the snapshot', () => {
    const { component } = setup()
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
