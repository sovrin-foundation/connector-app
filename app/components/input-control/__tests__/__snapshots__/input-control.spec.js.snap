// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletTabSendDetails /> should render properly and match the snapshot 1`] = `
<View
  accessibilityLabel="To-mainview-wallet-send-details"
  accessible={true}
  style={
    Object {
      "borderBottomColor": "#EAEAEA",
      "borderBottomWidth": 1,
    }
  }
  testID="To-mainview-wallet-send-details"
>
  <View
    accessibilityLabel="To-subview-wallet-send-details"
    accessible={true}
    center={true}
    fifth={true}
    horizontalSpace={true}
    row={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 15,
      }
    }
    testID="To-subview-wallet-send-details"
  >
    <View
      fifth={true}
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#ffffff",
          "paddingVertical": 25,
          "width": 35,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#333333",
            },
            Object {
              "fontWeight": "600",
            },
            null,
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "color": "rgba(237, 156, 46, 1)",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "lineHeight": 20,
            },
          ]
        }
      >
        To:
      </Text>
    </View>
    <View
      accessibilityLabel="To-textinput-wallet-send-details"
      accessible={true}
      fifth={true}
      horizontalSpace={true}
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "paddingHorizontal": 15,
          "width": "83%",
        }
      }
      testID="To-textinput-wallet-send-details"
    >
      <TextInput
        accessibilityLabel="To-token-send-details-label"
        accessible={true}
        multiline={false}
        onChangeText={[Function]}
        placeholder="test"
        style={
          Array [
            Object {
              "fontSize": 17,
              "fontStyle": "italic",
              "lineHeight": 20,
              "width": "100%",
            },
          ]
        }
        testID="To-token-send-details-label"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={Object {}}
    />
  </View>
</View>
`;
